### ISSUE TITLE
'check_leftovers' needs to be in sync with /usr/ports/Mk/Scripts/check_leftovers.sh

### ISSUE THREAD

**Issue by yurivict** [freebsd/poudriere#305]
In poudriere I am getting the errors about /usr/local/info/dir file being a "leftover".
After some poking around, I found these lines in /usr/ports/Mk/Scripts/check_leftovers.sh

                # removal of info files leaves entry uneasy to cleanup
                # in info/dir
                info/dir) ;;
                */info/dir) ;;

Since this problem is declared as "uneasy to cleanup" in ports, poudriere should probably just follow suit. Why doesn't it just include this Mk/Scripts/check_leftovers.sh and use the same logic?

Example of the problem which I hit: poudriere build fails for security/libassuan when NLS option is unset in /etc/make.conf. Unsetting NLS causes indexinfo not being pulled from dependences, and libassuan by itself fails to clean up this 'dir' file.

I personally don't believe this problem is really unsolvable. But for the time being it is better to have the same waiver in poudriere as in check_leftovers.sh.

I reported this to ports@ some time ago.

**Comment by bdrewery** [comment]
Poudriere uses check_leftovers.sh if it exists. I added that file to ports specifically so Poudriere could use it.

**Comment by yurivict** [comment]
So why 'info/dir' is failing as leftover then?

**Comment by bdrewery** [comment]
You didn't provide any logs. I don't know. The check_leftover.sh script also does _not_ allow info/dir to be a leftover. It only ignores modifications.

**Comment by yurivict** [comment]
@bdrewery  , see the log below. Failing condition is when the NLS option is unset in /etc/make.conf. This causes gettext-runtime to lose infoindex dependency, so libassuan 
is importing the indexinfo-0.2.3.txz package (see in the log).

Both post-install and post-deinstall scripts have "indexinfo ${file%25/*}". post-deinstall is supposed to delete 'dir' file.

It looks as if check_leftovers.sh is called before post-deinstall, not after post-deinstall, so it sees 'dir' file before it is deleted by indexinfo called from post-deinstall.

````
--- begin log ---
===========================================================================
====>> Checking for filesystem violations... done
=======================<phase: run-depends    >============================
===>   libassuan-2.2.0 depends on executable: indexinfo - not found
===>    Verifying install for indexinfo in /usr/ports/print/indexinfo
===>   Installing existing package /packages/All/indexinfo-0.2.3.txz
[local-system-10_1-STABLE-local-system-workstation-job-01] Installing indexinfo-0.2.3...
[local-system-10_1-STABLE-local-system-workstation-job-01] Extracting indexinfo-0.2.3... done
===>   Returning to build of libassuan-2.2.0
===========================================================================
====>> Recording filesystem state for prestage... done
=======================<phase: stage          >============================
===>  Staging for libassuan-2.2.0
===>   libassuan-2.2.0 depends on executable: indexinfo - found
===>   libassuan-2.2.0 depends on shared library: libgpg-error.so - found (/usr/local/lib/libgpg-error.so)
===>   Generating temporary packing list
if test -z 'strip'; then  /usr/bin/make  INSTALL_PROGRAM="/bin/sh /wrkdirs/usr/ports/security/libassuan/work/libassuan-2.2.0/build-aux/install-sh -c -s"  install_sh_PROGRAM="/bin/sh /wrkdirs/usr/ports/security/libassuan/work/libassuan-2.2.0/build-aux/install-sh -c -s" INSTALL_STRIP_FLAG=-s  install;  else  /usr/bin/make  INSTALL_PROGRAM="/bin/sh /wrkdirs/usr/ports/security/libassuan/work/libassuan-2.2.0/build-aux/install-sh -c -s"  install_sh_PROGRAM="/bin/sh /wrkdirs/usr/ports/security/libassuan/work/libassuan-2.2.0/build-aux/install-sh -c -s" INSTALL_STRIP_FLAG=-s  "INSTALL_PROGRAM_ENV=STRIPPROG='strip'" install;  fi
Making install in m4
Making install in src
/usr/bin/make  install-am
 /bin/mkdir -p '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/lib'
 /bin/sh ../libtool   --mode=install /usr/bin/install -c -o root -g wheel -s  libassuan.la '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/lib'
libtool: install: /usr/bin/install -c -o root -g wheel .libs/libassuan.so.0.5.0 /wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/lib/libassuan.so.0.5.0
libtool: install: strip --strip-unneeded /wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/lib/libassuan.so.0.5.0
libtool: install: (cd /wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/lib && { ln -s -f libassuan.so.0.5.0 libassuan.so.0 || { rm -f libassuan.so.0 && ln -s libassuan.so.0.5.0 libassuan.so.0; }; })
libtool: install: (cd /wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/lib && { ln -s -f libassuan.so.0.5.0 libassuan.so || { rm -f libassuan.so && ln -s libassuan.so.0.5.0 libassuan.so; }; })
libtool: install: /usr/bin/install -c -o root -g wheel .libs/libassuan.lai /wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/lib/libassuan.la
libtool: install: warning: remember to run `libtool --finish /usr/local/lib'
 /bin/mkdir -p '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/bin'
 install  -o root -g wheel -m 555 libassuan-config '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/bin'
/usr/bin/make  install-exec-hook
 /bin/mkdir -p '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/share/aclocal'
 install  -o root -g wheel -m 0644 libassuan.m4 '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/share/aclocal'
 /bin/mkdir -p '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/include'
 install  -o root -g wheel -m 0644 assuan.h '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/include'
Making install in doc
 /bin/mkdir -p '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/info'
 install  -o root -g wheel -m 0644 ./assuan.info '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/info'
 install-info --info-dir='/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/info' '/wrkdirs/usr/ports/security/libassuan/work/stage/usr/local/info/assuan.info'
Making install in tests
/usr/bin/make  install-am
====> Compressing man pages (compress-man)
===========================================================================
====> Running Q/A tests (stage-qa)
====> Checking for pkg-plist issues (check-plist)
===> Parsing plist
===> Checking for items in STAGEDIR missing from pkg-plist
===> Checking for items in pkg-plist which are not in STAGEDIR
===> No pkg-plist issues found (check-plist)
====>> Checking for staging violations... done
=======================<phase: package        >============================
===>  Building package for libassuan-2.2.0
===========================================================================
=======================<phase: install-mtree  >============================
===========================================================================
====>> Recording filesystem state for preinst... done
=======================<phase: install        >============================
===>  Installing for libassuan-2.2.0
===>   libassuan-2.2.0 depends on executable: indexinfo - found
===>   libassuan-2.2.0 depends on shared library: libgpg-error.so - found (/usr/local/lib/libgpg-error.so)
===>  Checking if libassuan already installed
===>   Registering installation for libassuan-2.2.0
[local-system-10_1-STABLE-local-system-workstation-job-01] Installing libassuan-2.2.0...
===> SECURITY REPORT: 
      This port has installed the following files which may act as network
      servers and may therefore pose a remote security risk to the system.
/usr/local/lib/libassuan.so.0.5.0

      If there are vulnerabilities in these programs there may be a security
      risk to the system. FreeBSD makes no guarantee about the security of
      ports included in the Ports Collection. Please type 'make deinstall'
      to deinstall the port if this is a concern.

      For more information, and contact details about the security
      status of this software, see the following webpage: 
http://gnupg.org/related_software/libassuan/index.en.html
===========================================================================
====>> Checking shared library dependencies
 0x0000000000000001 (NEEDED)             Shared library: [libc.so.7]
 0x0000000000000001 (NEEDED)             Shared library: [libgpg-error.so.0]
=======================<phase: deinstall      >============================
===>  Deinstalling for libassuan
===>   Deinstalling libassuan-2.2.0
Updating database digests format... done
Checking integrity... done (0 conflicting)
Deinstallation has been requested for the following 1 packages (of 0 packages in the universe):

Installed packages to be REMOVED:
	libassuan-2.2.0

The operation will free 277 KiB.
[local-system-10_1-STABLE-local-system-workstation-job-01] [1/1] Deinstalling libassuan-2.2.0...
[local-system-10_1-STABLE-local-system-workstation-job-01] [1/1] Deleting files for libassuan-2.2.0... done
===========================================================================
====>> Checking for extra files and directories
====>> Error: Files or directories left over:
info/dir
====>> Cleaning up wrkdir
===>  Cleaning for libassuan-2.2.0
build of security/libassuan ended at Mon May  4 14:13:31 PDT 2015
build time: 00:00:11
!!! build failure encountered !!!
````

````
{
  "post-install":"/sbin/ldconfig -m /usr/local/lib\n  case \"/usr/local/info/assuan.info\" in\n  /*) file=\"/usr/local/info/assuan.info\" ;;\n   *) file=\"/usr/local//usr/local/info/assuan.info\" ;;\n  esac\n  indexinfo ${file%25/*}",
  "post-deinstall":"/sbin/ldconfig -R\n  case \"/usr/local/info/assuan.info\" in\n  /*) file=\"/usr/local/info/assuan.info\" ;;\n   *) file=\"/usr/local//usr/local/info/assuan.info\" ;;\n  esac\n  indexinfo ${file%25/*}"
}}
````

**Comment by bdrewery** [comment]
It is an actual leftover. The post-deinstall is not trying to remove it either. This is not a poudriere problem. It is a ports problem.

**Comment by bdrewery** [comment]
I'll look into it anyhow.

**Comment by yurivict** [comment]
I found another bug:
libgpg-error installs .info w/out INFO= statement.
This might be what triggers the problem.

**Comment by yurivict** [comment]
This patch https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=199975 needs to be checked in

**Comment by yurivict** [comment]
@bdrewery  ,
Please commit this new stage-qa test 'infoplist': https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=199978

It will close this problem for good.
Some ports will fail to build in stage-qa, they should be fixed individually, like the above bug#199975.
